File: src/main/java/github/macro/Launcher.java
package github.macro;

import github.macro.ui.selector.SelectorController;
import github.macro.ui.selector.SelectorModel;
import github.macro.ui.selector.SelectorView;
import javafx.application.Application;
import javafx.scene.Scene;
import javafx.stage.Stage;
import org.apache.logging.log4j.Level;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.util.Arrays;

/**
 * Created by Macro303 on 2019-Nov-29.
 */
public class Launcher extends Application {
	private static final Logger LOGGER = LogManager.getLogger(Launcher.class);

	public static void main(String[] args) {
		checkLogLevels();
		LOGGER.info("Initializing Exile Buddy");
		launch(args);
	}

	private static void checkLogLevels() {
		Arrays.stream(Level.values()).sorted().forEach(level -> LOGGER.log(level, "{} is Visible", level.name()));
	}

	@Override
	public void start(Stage primaryStage) {
		SelectorModel model = new SelectorModel();
		SelectorController controller = new SelectorController(model);
		SelectorView view = new SelectorView(controller, model);

		Scene scene = new Scene(view.asParent(), 400, 400);
		scene.getStylesheets().add(getClass().getResource("Dark-Theme.css").toExternalForm());
		scene.getStylesheets().add(getClass().getResource("Custom.css").toExternalForm());
		primaryStage.setScene(scene);
		primaryStage.show();
	}
}

File: src/main/kotlin/github/macro/Launcher.kt
package github.macro

import github.macro.config.Config
import github.macro.ui.Selector
import javafx.scene.text.FontWeight
import org.apache.logging.log4j.Level
import org.apache.logging.log4j.LogManager
import tornadofx.*
import java.util.*

/**
 * Created by Macro303 on 2020-Jan-13.
 */
class Launcher : App(Selector::class, Styles::class) {
	init {
		checkLogLevels()
		LOGGER.info("Welcome to Exile Buddy")
		if (Config.INSTANCE.useDarkMode) {
			importStylesheet(Launcher::class.java.getResource("Modena-Dark.css").toExternalForm())
//		    importStylesheet(Launcher::class.java.getResource("Custom-Dark.css").toExternalForm())
		}
		FX.locale = Locale.ENGLISH
		reloadStylesheetsOnFocus()
	}

	private fun checkLogLevels() {
		Level.values().sorted().forEach {
			LOGGER.log(it, "{} is Visible", it.name())
		}
	}

	companion object {
		private val LOGGER = LogManager.getLogger(Launcher::class.java)
	}
}

class Styles : Stylesheet() {
	init {
		title {
			fontSize = 35.pt
			fontWeight = FontWeight.BOLD
		}
	}

	companion object {
		private val LOGGER = LogManager.getLogger(Styles::class.java)
		val title by cssid()
	}
}